src/tests/integration/client.test.ts

```diff
@@ -119,4 +121,71 @@ describe('client', () => {
       expect(details).toHaveProperty('id', id);
     });
   });
+
+  describe('tunes', () => {
+    test('should list all tunes using callbacks', () =>
+      new Promise((done) => {
+        expect.assertions(tunesStore.length * 2);
+        let count = tunesStore.length;
+        client.tunes((err, tune) => {
+          expect(err).toBeFalsy();
+          expect(tune).toBeDefined();
+          if (--count === 0) done(undefined);
+        });
+      }));
+
+    test('should list all tunes', async () => {
+      const allTunes = [];
+      for await (const tune of client.tunes()) {
+        allTunes.push(tune);
+      }
+      expect(allTunes.length).toBe(tunesStore.length);
+      allTunes.forEach((tune, idx) => {
+        expect(tune).toHaveProperty('id', tunesStore[idx].id);
+      });
+    });
+
+    test('should list all tune methods', async () => {
+      const tuneMethods = await client.tuneMethods();
+      expect(tuneMethods.length).toBe(tuneMethodsStore.length);
+    });
+
+    test('should show details of a tune', async () => {
+      const { id } = tunesStore[0];
+      const tune = await client.tune({ id });
+      expect(tune).toHaveProperty('id', id);
+    });
+
+    test('should download assets of a completed tune', async () => {
+      const {
+        id,
+        assets: { encoder },
+      } = tunesStore[0];
+      const tune = await client.tune({ id });
+      expect.assertions(1);
+      if (tune.status === 'COMPLETED') {
+        const content = await tune.downloadAsset('encoder');
+        // unfortunately, we don't have .toArray() in the supported version of Node
```

Comment:
Soon we will have [fromAsync](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fromAsync) 👍🏻

Reply:
yeah sorry, `fromAsync` 👍